llama_hub/llama_packs/fusion_retriever/hybrid_fusion/base.py

```diff
@@ -19,19 +18,39 @@ class HybridFusionRetrieverPack(BaseLlamaPack):
     """
 
     def __init__(
-        self,
-        nodes: List[TextNode] = None,
-        chunk_size: int = 256,
-        mode: str = "reciprocal_rerank",
-        vector_similarity_top_k: int = 2,
-        bm25_similarity_top_k: int = 2,
-        fusion_similarity_top_k: int = 2,
-        num_queries: int = 4,
-        **kwargs: Any,
+            self,
+            nodes: List[TextNode] = None,
+            chunk_size: int = 256,
+            mode: str = "reciprocal_rerank",
+            vector_similarity_top_k: int = 2,
+            bm25_similarity_top_k: int = 2,
+            fusion_similarity_top_k: int = 2,
+            num_queries: int = 4,
+            documents: List[Document] = None,
+            cache_dir: str = None,
+            **kwargs: Any,
     ) -> None:
         """Init params."""
         service_context = ServiceContext.from_defaults(chunk_size=chunk_size)
-        index = VectorStoreIndex(nodes, service_context=service_context)
+        if cache_dir is not None and os.path.exists(cache_dir):
+            # Load from cache
+            from llama_index import StorageContext, load_index_from_storage
+            # rebuild storage context
+            storage_context = StorageContext.from_defaults(
+                persist_dir=cache_dir)
+            # load index
+            index = load_index_from_storage(storage_context)
+        elif documents is not None:
+            index = VectorStoreIndex.from_documents(
+                documents=documents, service_context=service_context
+            )
+            # Cache the index
+            index.storage_context.persist()
+        else:
+            index = VectorStoreIndex(nodes, service_context=service_context)
+
+        if cache_dir is not None:
```

Comment:
given the two above comments let's remove this

Reply:
actually, this line is supposed to be as a generalization, instead of duplicating the cache. I'll push a version with only this line, if you want me to go with the duplicated version please LMK